### PR TITLE
Feature/localisation

### PR DESCRIPTION
**Pull request by IhToN**
Add a Localisation system.

Localise.jsx:
Base component for such task.
Available props:

1. node _String_ - translation node using dot notation: "buttons.cancel"
2. variables _Object_ - JSON Object being key the variable identifier: "{{total}} decks" -> { total: Math.random() }
3. dangerouslySetInnerHTML _Boolean_ - Set it to true if the translation includes html code that has to be parsed

Public functions:

1. fetchTranslation(langCode = 'es') - load a translation file from the /locales/ folder
2. parseTranslation(nodeString, variables, translations) - retrieve the translated string instead of a JSX

### CODE REVIEW

File: www/src/pages/NewGame.js [hunk 1]
```diff
@@ -131,12 +132,12 @@ export default function NewGame ({ navigate }) {
   const [decksTree] = useDecks()
   const decks = Object.values(decksTree)
 
-  const rotationOptions = config.rotationOptions
+  const rotationOptions = config.rotationOptions.map(item => Object.assign({}, item, {label: <Localise node={item.label} />}))
```
Comment (juandjara) [inline www/src/pages/NewGame.js:135]: Small thing: Prefer `{ ...o1, ...o2 }` notation over `Object.assign({}, o1, o2)`

File: www/src/components/Spinner.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import spinner from '../assets/spinner.svg';
+import styled from "styled-components";
+
+const SpinnerStyles = styled.div`
+    display: flex;
```
Comment (juandjara) [inline www/src/components/Spinner.js:6]: Use "2 Spaces" indentation on all files, please

File: www/src/components/Localise.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import useGlobalSlice from '../services/useGlobalSlice'
+
+const nodeReducer = (object, key) => object[key] || '';
+
+const getTranslation = (nodeString, translations) => {
+    const node = nodeString.split('.');
```
Comment (juandjara) [inline www/src/components/Localise.js:7]: Use "2 Spaces" indentation on all files, please

File: www/src/App.js [hunk 4]
```diff
@@ -23,7 +27,46 @@ const StyledRoot = styled(Router)`
...
+
+  const languageFromLS = localStorage.getItem(config.LANGUAGE_KEY);
+  let fallbackLanguage = config.availableLanguages[0];
+  try {
+    fallbackLanguage = JSON.parse(languageFromLS) || fallbackLanguage;
+  } catch (ignore) {}
+
+  useEffect(() => {
+    if(language) {
+      fetchTranslation(language.value).then(translation => {
```
Comment (juandjara) [inline www/src/App.js:48]: Prefer `async / await` with a `try / catch` block over `Promise.then` and `Promise.catch`

**Review by juandjara — CHANGES_REQUESTED** [review]
Everything is fine. Just added some small comments regarding to code style.
Maybe I would the translation related to something like a `useTranslation` hook
This way you would have something like
```
const [parseTranslations, language, setLanguage] = useTranslations()
```
or something like that. 

Also it would be fine if you checked `navigator.language` in order to set the default translation

But this type of structural changes can be reviewed in a following iteration of this feature

**Review by juandjara — APPROVED** [review]

**Comment by juandjara** [comment]
Well, everything seems fine now 💯 
@IhToN can you please check the "in-game" screen to check if there is anything left to translate?

**Comment by IhToN** [comment]
I've changed the Localise component. It is now a hook ;-)

Gonna check the InGame screen.

**Pull request merged**